core/server/src/main/java/alluxio/worker/block/BlockMetadataManagerView.java

```diff
@@ -229,4 +229,19 @@ public BlockMeta getBlockMeta(long blockId) throws BlockDoesNotExistException {
       return null;
     }
   }
+
+  /**
+   * Return -1 if block does not exist, else return the byte size of the block.
+   *
+   * @param blockId id of the block
+   * @return block size or -1 if the block does not exist
+   * @throws BlockDoesNotExistException if no {@link BlockMeta} for this block id is found
+   */
+  public long getBlockSizeIfExist(long blockId) throws BlockDoesNotExistException {
+    if (mMetadataManager.hasBlockMeta(blockId)) {
+      return mMetadataManager.getBlockMeta(blockId).getBlockSize();
```

Comment:
@uronce-cc `getBlockSizeIfExist` is used in `onCommitBlock` of `LIRSEvictor`.  I check the code and find that `mMetadataManager.getBlockMeta(blockId)` will return null if the block is pinned, locked or marked, but the block size is important for `LIRSEvictor` and has to be known.  I have met with the errors caused by this issue, so we add `getBlockSizeIfExist` here